### DISCUSSION TITLE
Styling based on field value

### DISCUSSION THREAD

**Discussion by lsmoker** [jrief/django-formset#124] [Q&A]
Is there a preferred way to style/add a class to a field based on that field's value? I know I could create widgets for each one, but that becomes tedious.

**Comment by jrief** [comment]
> Is there a preferred way to style/add a class to a field based on that field's value?

You mean field name? Yes there is a way, this example uses different CSS classes for each field:
https://django-formset.fly.dev/bootstrap/address

Here I pass
```
form_css_classes = 'row'
field_css_classes = {
        '*': 'mb-2 col-12',
        'postal_code': 'mb-2 col-4',
        'city': 'mb-2 col-8',
}
```
to the Bootstrap form renderer. Other frameworks use other classes.

This is documented here: https://django-formset.fly.dev/form-renderer/

**Comment by lsmoker** [comment]
Actually, I do mean "value". What I would like to do (for one example) is change the color of the text based on whether or not it is negative, so while the default color (I've used the css_classes you mentioned) might be green or black, I want it to be red if the value (dollar amount) is negative.

**Reply by jrief** [reply]
This would be a nice use-case for a directive similar to `df-show`/`df-hide` such as
 `df-class("my-css-class")="<some_formset_condition>"`.

I'll check, if that can be implemented.

**Reply by lsmoker** [reply]
Another use case would be to set the "title" attribute to the value (so it shows when the field control is too short (esp. if the field is disabled)) on a field to get hover text.
